tools/docs/config.ts

```diff
@@ -142,13 +145,64 @@ function genTable(obj: [string, string][], type: string, def: any): string {
 }
 
 function stringifyArrays(el: Record<string, any>): void {
+  const ignoredKeys = ['default', 'experimentalIssues'];
+
   for (const [key, value] of Object.entries(el)) {
-    if (key !== 'default' && Array.isArray(value)) {
+    if (!ignoredKeys.includes(key) && Array.isArray(value)) {
       el[key] = value.join(', ');
     }
   }
 }
 
+function genExperimentalMsg(el: Record<string, any>): string {
+  const ghIssuesUrl = 'https://github.com/renovatebot/renovate/issues/';
+  let warning =
+    '\n<!-- prettier-ignore -->\n!!! warning "This feature is flagged as experimental"\n';
+
+  if (!el.experimental) {
+    return '';
+  }
+
+  if (el.experimentalDescription) {
+    warning += indent`${2}${el.experimentalDescription}`;
+  } else {
+    warning += indent`${2}Experimental features might be changed or even removed at any time.`;
+  }
+
+  const issues: string[] = [];
+  for (const issue of el.experimentalIssues ?? []) {
+    issues.push(`[#${issue}](${ghIssuesUrl}${issue})`);
+  }
+
+  if (issues.length > 0) {
+    warning += `<br>To track this feature visit the following GitHub ${
+      issues.length > 1 ? 'issues' : 'issue'
+    } `;
+    warning += issues.join(', ') + '.\n';
+  }
+
+  return warning;
+}
+
+function indexMarkdown(lines: string[]): Record<string, [number, number]> {
+  const indexed: Record<string, [number, number]> = {};
+
+  let optionName = '';
+  let start = 0;
+  for (const [i, line] of lines.entries()) {
+    if (line.startsWith('##')) {
```

Comment:
```suggestion
    if (line.startsWith('## ') || line.startsWith('### ')) {
```

We've also `####`, which aren't options but sub header. Not sure if we should render the warning before or after those 🤔
@rarkins WDYT? 

https://github.com/renovatebot/renovate/blob/8ce54366e169daf62749267d78633dce540ad845/docs/usage/configuration-options.md#L2368